### PR TITLE
Logging improvements

### PR DESCRIPTION
**Pull request by uthsab**
- ensure python subprocess errors are printed. 
- log to error, info, degug files along with printing on console. will help in log collection.

### CODE REVIEW

**Comment by msfttusharjuneja** [comment]
Will it make sense to have info, error and debug (or at least debug) always getting created? The intention behind this thought is that the cx will not have to re-run with LogLevel set to a higher level for us to catch the issue and this way we will have context saved for each failed run.

File: src/appliance-onboarding-script/appliance_setup/run.py [hunk 1]
```diff
@@ -18,6 +18,43 @@
 from avs.avsarconboarder.deleter.arcadon.arc_addon_deleter import ArcAddonDeleter
 from pkgs._utils import confirm_prompt
 
+def logger_setup(logLevel = logging.INFO):
```
Comment (msfttusharjuneja) [inline src/appliance-onboarding-script/appliance_setup/run.py:21]: Should we create a separate folder for these log files? Also does it make sense to keep log files for all the onboarding attempts? Something like log_<Current_Timestamp>?

File: src/appliance-onboarding-script/appliance_setup/avs/avsarconboarder/executor/process/process_executor.py [hunk 2]
```diff
@@ -21,6 +23,11 @@ def execute(self, *args):
...
         else:
             process_cmd = self._process_cmd
-        output = subprocess.check_output(process_cmd, timeout=100000, shell=True)
-        result = output.decode('UTF-8', errors='strict')
+        try:
+            output = subprocess.check_output(process_cmd, timeout=100000, shell=True)
+        except subprocess.CalledProcessError as e:
+            logging.error(bytes_to_string(e.output))
+            return None
+        result = bytes_to_string(output)
```
Comment (msfttusharjuneja) [inline src/appliance-onboarding-script/appliance_setup/avs/avsarconboarder/executor/process/process_executor.py:31]: Check if result can be computed first and returned as part of exception handling also.

File: src/appliance-onboarding-script/appliance_setup/run.py (hunk 1 above)
Comment (uthsab) [reply src/appliance-onboarding-script/appliance_setup/run.py:22]: updated

**Comment by uthsab** [comment]
> Will it make sense to have info, debug always getting created? The intention behind this thought is that the cx will not have to re-run with LogLevel set to a higher level for us to catch the issue and this way we will have context saved for each failed run.

As discussed, the ideal state would be to log info as default level, and log debug only when required. Our logs should be refined enough to fit this. That was not the state so far. We have introduced the change to log az command errors correctly. We can continue to have info as the default level and see the onboarding experience for now. We can revisit this call if we see info and above logs being insufficient.

**Review by msfttusharjuneja — APPROVED** [review]
I am fine with these changes. Did you get a chance to analyze how much are the changes for spitting out exceptions rather than returning error codes?

**Comment by uthsab** [comment]
> I am fine with these changes. Did you get a chance to analyze how much are the changes for spitting out exceptions rather than returning error codes?

That would involve changing the portion of code we did not initially own and it is at an infra level, the impact will tough multiple paths. We can pick up that refactoring task separately when necessary. It is fine to not include the change at this time.

**Pull request merged**